File: postgresql/nextjs/src/repositories/test1-repository.ts
import { PoolClient } from 'pg';
import { Test1 } from '../models/entities/test1';

export class Test1Repository {
    static async create(client: PoolClient, test1: Test1) {
        const result = await client.query("INSERT INTO test1(test) VALUES($1) RETURNING id;", [test1.test])
        return result
    }

    static async getById(client: PoolClient, id: number) {
        const result = await client.query("SELECT id, test FROM test1 WHERE id = $1;", [id])
        return result
    }

    static async update(client: PoolClient, test1: Test1) {
        const result = await client.query("UPDATE test1 SET test = $1 WHERE id = $2;", [test1.test, test1.id])
        return result
    }

    static async delete(client: PoolClient, id: number) {
        const result = await client.query("DELETE FROM test1 WHERE id = $1;", [id])
        return result
    }
}

File: mysql/net9/Repositories/Test1Repository.cs
using System.Data;
using MySql.Data.MySqlClient;

namespace net9.Repositories {

    public class Test1Repository: ITest1Repository
    {
        // public async Task<long> Create(IDbConnection db, Test1 test1, IDbTransaction transaction)
        // {
        //     return await db.ExecuteScalarAsync<long>("INSERT INTO test1(test) VALUES(@Test); SELECT LAST_INSERT_ID();", new {Test = test1.Test}, transaction: transaction);
        // }

        // public async Task<Test1?> GetById(IDbConnection db, int id) 
        // {
        //     return await db.QueryFirstOrDefaultAsync<Test1>("SELECT id, test FROM test1 WHERE id = @Id", new {Id = id});
        // }

        // public async Task<int> Update(IDbConnection db, Test1 test1)
        // {
        //     return await db.ExecuteAsync("UPDATE test1 SET test = @Test WHERE id = @Id", new {Id = test1.Id, Test = test1.Test});
        // }

        // public async Task<int> Delete(IDbConnection db, int id)
        // {
        //     return await db.ExecuteAsync("DELETE FROM test1 WHERE id = @Id", new { Id = id });
        // }

        public async Task<Test1?> GetById(MySqlConnection connection, int id)
        {
            using var command = new MySqlCommand("SELECT id, test FROM test1 WHERE id = @Id", connection);
            command.Parameters.AddWithValue("@Id", id);
            using var reader = await command.ExecuteReaderAsync();
            if (await reader.ReadAsync())
            {
                return new Test1
                {
                    Id = reader.GetInt32(0),
                    Test = reader.GetString(1)
                };
            }
            return null;
        }
    }

}